sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/execution_dispatcher.py

```diff
@@ -107,12 +107,27 @@ async def _create_pipelined_execution_context(self, query_execution_info):
                 raise CosmosHttpResponseError(StatusCodes.BAD_REQUEST,
                                   "Cross partition query only supports 'VALUE <AggregateFunc>' for aggregates")
 
-        execution_context_aggregator = multi_execution_aggregator._MultiExecutionContextAggregator(self._client,
+        # throw exception here for vector search query without limit filter
+        if query_execution_info.get_non_streaming_order_by():
+            if query_execution_info.get_top() is None and query_execution_info.get_limit() is None:
+                # TODO: missing one last if statement here to check for the system variable bypass - need name
+                raise ValueError("Executing a vector search query without TOP or LIMIT can consume many" +
```

Comment:
For Cosmos DB, LIMIT cannot be used separately, it has to be combined with OFFSET. It will be in best interest of customer to clarify this here. Or rather just have the top as the query constraint?
@xinlian12 / @FabianMeiswinkel  thoughts?

Reply:
queries will already throw an error when you use LIMIT without OFFSET as a separate error so there shouldn't be a need to check for both again here, I definitely can if you want me to though

Reply:
OFFSET and LIMIT is a very expensive construct thought - effectively need to get OFFSET + LIMIT records (TOP == OFFSET + LIMIT) due to the backend not being able to do the skip effectively - so, I am ok with limiting to TOP - if you include LIMIT you probably have to change the validation to OFFSET + LIMIT < MaxBufferedItemCount